### ISSUE TITLE
[Z-wave core] S2 inclusion incomplete - missing command classes - unable to register capabilities

### ISSUE THREAD

**Issue by TedTolboom** [athombv/apps-sdk-issues#56]
When trying to include several (certified) S2 devices, inclusion / interview of supported command_classes (specifically of multichannel nodes) is incomplete or not matching new S2 standards.

Example is the Heatit Z-TRM2FX: https://products.z-wavealliance.org/Search/ProductDetails?productId=3065

Which has on MultiChannelNode 2-4 the individual temperature sensor representations, as reflected in by `DeviceClassGeneric: GENERIC_TYPE_SENSOR_MULTILEVEL`. 

```
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] - MultiChannelNode: 2
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] - DeviceClassGeneric: GENERIC_TYPE_SENSOR_MULTILEVEL
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] -- CommandClass: COMMAND_CLASS_ZWAVEPLUS_INFO
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] --- Version: 2
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] --- Commands:
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- ZWAVEPLUS_INFO_GET
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- ZWAVEPLUS_INFO_REPORT
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] -- CommandClass: COMMAND_CLASS_SUPERVISION
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] --- Version: 1
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] --- Commands:
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SUPERVISION_GET
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SUPERVISION_REPORT
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] -- CommandClass: COMMAND_CLASS_SECURITY
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] --- Version: 1
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] --- Commands:
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- NETWORK_KEY_SET
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- NETWORK_KEY_VERIFY
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_COMMANDS_SUPPORTED_GET
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_COMMANDS_SUPPORTED_REPORT
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_MESSAGE_ENCAPSULATION
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_MESSAGE_ENCAPSULATION_NONCE_GET
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_NONCE_GET
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_NONCE_REPORT
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_SCHEME_GET
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_SCHEME_INHERIT
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_SCHEME_REPORT
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] -- CommandClass: COMMAND_CLASS_SECURITY_2
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] --- Version: 1
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] --- Commands:
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_2_NONCE_GET
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_2_NONCE_REPORT
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_2_MESSAGE_ENCAPSULATION
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- KEX_GET
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- KEX_REPORT
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- KEX_SET
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- KEX_FAIL
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- PUBLIC_KEY_REPORT
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_2_NETWORK_KEY_GET
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_2_NETWORK_KEY_REPORT
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_2_NETWORK_KEY_VERIFY
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_2_TRANSFER_END
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_2_COMMANDS_SUPPORTED_GET
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_2_COMMANDS_SUPPORTED_REPORT
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_2_CAPABILITIES_GET
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- SECURITY_2_CAPABILITIES_REPORT
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] -- CommandClass: COMMAND_CLASS_BASIC
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] --- Version: 1
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] --- Commands:
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- BASIC_GET
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- BASIC_REPORT
2019-01-03 11:11:35 [log] [ManagerDrivers] [Z-TRM2fx] [0] ---- BASIC_SET
```
The actual command class, Multilevel Sensor (version 5), is however not reflected in the node overview. Based on the Z-wave sniffer (Zniffer), it is clear that the device does report the temperatures from these multi-channel nodes.

Since it is not available in the node overview, the Z-wave meshdriver does not allow to register any listeners to it and there is not option to override the Z-wave meshdriver checks. 

Please make sure that the inclusion of a S2 device is done according to the Z-wave specification and adopt the meshdriver accordingly. And / Or provide an option within the meshdriver to override the check if the command class is available.

Attached the inclusion logs (Zniffer) and printNodes for the Heatit Z-TRM2FX, Z-Water and Z-wave thermostat (FW 1.92 = S2). 
[Z-TRM2fx.zip](https://github.com/athombv/homey-apps-sdk-issues/files/2872725/Z-TRM2fx.zip)
[Z-Water.zip](https://github.com/athombv/homey-apps-sdk-issues/files/2872726/Z-Water.zip)
[Z-wave FW1.92.zip](https://github.com/athombv/homey-apps-sdk-issues/files/2872727/Z-wave.FW1.92.zip)

**Comment by WeeJeWel** [comment]
It seems like the device is not telling Homey which command classes it has during pairing, because Homey does not support S2 yet.

I don't think there's a workaround available at this point but wait for S2 to work.

**Comment by WeeJeWel** [comment]
@TedTolboom What you can try: create a Diagnostics Report after the device has reported a value that Homey didn't parse. Maybe I can forward it to the right app.

**Comment by TedTolboom** [comment]
@WeeJeWel see Diagnostics Report: 0675542B30

After the device reported through:
<img width="684" alt="schermafbeelding 2019-03-04 om 14 27 40" src="https://user-images.githubusercontent.com/17163431/53736339-b59e4180-3e89-11e9-8177-a2463dc0ce37.png">

**Comment by WeeJeWel** [comment]
In Homey v2.0.5-rc.3, a ZwaveNode can emit an `unknownReport` event where the first parameter is a `Buffer` with the entire payload.

```
this.node.on('unknownReport', buf => {
   const commandClassId = buf[0];
})
```

**Comment by WeeJeWel** [comment]
Fixed in Homey v2.0.5-rc.5 for both root and multichannel nodes.
